[lshaoqin] iP

lshaoqin opened the pull request:
# JukeBot

> "Best juke box I have ever seen." - Shaoqin ([source](https://www.youtube.com/watch?v=dQw4w9WgXcQ))

JukeBot frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ JUKER FAST to use

All you need to do is,

download it from [here](https://github.com/lshaoqin/ip).
1. double-click it.
2. add your tasks.
3. let it manage your tasks for you 😉
And it is **FREE**!

Features:

- [x] Managing jukes
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
___
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

lshaoqin committed:
implement skeletal version

lshaoqin committed:
add welcome and goodbye messages

lshaoqin committed:
add ability to echo and exit

lshaoqin committed:
add and list functions

lshaoqin committed:
mark as done and undone

lshaoqin committed:
todos, events, deadlines

lshaoqin committed:
add tests

lshaoqin committed:
exception handling

lshaoqin committed:
exception handling

lshaoqin committed:
implement delete function

lshaoqin committed:
Implement saving feature

lshaoqin committed:
Merge branch 'branch-Level-7'

lshaoqin committed:
Add datetime support

lshaoqin committed:
Merge branch 'branch-Level-8'

lshaoqin committed:
refactor based on OOP

lshaoqin committed:
organise into package

lshaoqin committed:
Merge remote-tracking branch 'origin/add-gradle-support'

lshaoqin committed:
Add JUnit tests

lshaoqin committed:
Add JAR support

lshaoqin committed:
Add Javadoc comments

lshaoqin committed:
Style code based on coding standards

lshaoqin committed:
Implement find function

lshaoqin committed:
Merge branch 'branch-A-CodingStandard'

lshaoqin committed:
Merge branch 'branch-Level-9'

lshaoqin committed:
Update README.md

lshaoqin committed:
Update README.md

lshaoqin committed:
add checkstyle

lshaoqin committed:
Merge branch 'master' of https://github.com/lshaoqin/ip

lshaoqin committed:
implement GUI

aexolate commented on src/main/java/juke/Parser.java:
```diff
@@ -0,0 +1,83 @@
+     * @param input Input from the user.
+     * @throws JukeError If there was a problem parsing the input.
+     */
+    String parse(String input) throws Exception {
```
Parse method is neat and does not have deep nesting

aexolate commented on src/main/java/juke/Parser.java:
```diff
@@ -0,0 +1,83 @@
+        }
+        else if (input.startsWith("event")) {
+            final Pattern eventPattern = Pattern.compile(
+                    "^event\\s+(.*)\\s+/from\\s+(.*)\\s+/to\\s+(.*)$");
```
Regex is kind of cryptic, perhaps a short comment explaining the expression can help other readers?

aexolate commented on build.gradle:
```diff
@@ -0,0 +1,68 @@
+}
+
+application {
+    mainClass.set("juke.Juke")
```
I think this should be changed to juke.Launcher or gradle won't work properly

aexolate commented on src/main/java/juke/TaskList.java:
```diff
@@ -0,0 +1,57 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    public ArrayList<Task> tasks;
```
Should tasks be made private?

aexolate reviewed (commented):
LGTM. The UI looks great and the application seems to be working well. The code could help with more comments and documentation.

lshaoqin committed:
make changes according to pr review

lshaoqin commented:
Thank you for the comments! I have made the appropriate changes.

Mahidharah commented on src/main/java/juke/Ui.java:
```diff
@@ -0,0 +1,102 @@
+     * Prints the message to welcome the user.
+     */
+    public static String printStart() {
+        //Introduce itself to the user
```
Should this comment be here? Maybe add it to the javadocs instead?

Mahidharah commented on src/main/java/juke/Ui.java:
```diff
@@ -0,0 +1,102 @@
+     * Prints bye to the user.
+     */
+    public String printBye() {
+        //Say goodbye
```
Same here, might be less cluttered if shifted to a javadoc comment

Mahidharah commented on src/main/java/juke/Storage.java:
```diff
@@ -0,0 +1,105 @@
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class Storage {
```
No javadoc for class? Maybe a short description of what the class does might be helpful!

Mahidharah commented on src/main/java/juke/Task.java:
```diff
@@ -0,0 +1,51 @@
+package juke;
+
+public class Task {
+    protected final String desc;
```
Consider naming the full name "description" for better readability.

Mahidharah reviewed (commented):
Just some nits to fix! But overall really well done! Looks better than mine :P

lshaoqin committed:
Add assertions to the main Juke method

Assertions help to ensure that the Juke methods are
changing the tasklist as they are supposed to.

To find possible bugs in the code, let's define some
assertions to identify where things aren't going as
planned.

lshaoqin commented:
Thank you! I have made the appropriate changes.

lshaoqin committed:
Improve code quality

lshaoqin committed:
Merge branch 'branch-A-Assertions'

lshaoqin committed:
Merge branch 'branch-A-CodeQuality'

lshaoqin committed:
Add CI

lshaoqin committed:
Add tagging functionality

lshaoqin committed:
Style changes

lshaoqin committed:
Merge branch 'branch-C-Tagging'

lshaoqin committed:
Add readme

lshaoqin committed:
style changes

lshaoqin committed:
Merge pull request #2 from lshaoqin/A-Codestyle

Style code and import sequences

lshaoqin committed:
Change Ui.png name
